library/src/scala/tasty/Reflection.scala

```diff
@@ -362,7 +362,17 @@ trait Reflection { reflection =>
 
   val Term: TermModule
 
-  trait TermModule { this: Term.type => }
+  trait TermModule { this: Term.type =>
+
+    /** Returns a term that is functionally equivalent to `t`,
+     *  however if `t` is of the form `((y1, ..., yn) => e2)(x1, ..., xn)`
+     *  then it optimizes this the top most call by returning the `Some`
+     *  with the result of beta-reducing the application.
+     *  Otherwise returns `None`.
```

Comment:
Will it reduce if `x1, ..., xn` are constants?

Reply:
Yes, but it will introduce bindings to preserve the semantics all our beta-reduction optimization do (`Expr.betaReduce` and `BetaReduce` miniphase). They all use the same logic internally.

Reply:
Then it might be better to use `e1, ..., en` instead of `x1, ..., xn` and say explicitly that it will preserve semantics by introduce bindings.

Reply:
I added the extra documentation